Tensile/KernelWriter.py

```diff
@@ -104,8 +104,32 @@ def makeSchedule(self, kernel, tensorParametersA, tensorParametersB, localWriteE
     lastLoadIter = 0
     if kernel["EnableMatrixInstruction"] and kernel["ScheduleIterAlg"] == 3:
       numMfmaPerIter = kernel["MIWaveTile"][0] * kernel["MIWaveTile"][1] * kernel["InnerUnroll"]
+      # Can locally overrid these ######
+      # number of mfma between last localWrite and barrier
+      numMfmaBetweenLWandBarrier = 1 if kernel["MatrixInstM"] == 32 else 2
+      # number of global read instructions between 2 mfma
+      self.numGlobalReadInsPerMfma = 2 if kernel["MatrixInstM"] == 32 and not kernel["ProblemType"]["TLUA"] and not kernel["ProblemType"]["TLUB"] and kernel["TransposeLDS"] and not kernel["1LDSBuffer"] else 1
+      MatrixInstructionLatency = kernel["MatrixInstM"] // 2 - 1
+      LocalWriteLatency = tensorParametersA["localWriteInstruction"].IssueLatency*2
+      numReadPerVectorA = tensorParametersA["bpe"] * kernel["LocalReadVectorWidth"] // int(tensorParametersA["localReadInstruction"].blockWidth * 4)
+      numReadPerVectorB = tensorParametersB["bpe"] * kernel["LocalReadVectorWidth"] // int(tensorParametersB["localReadInstruction"].blockWidth * 4)
+      numA = kernel["InnerUnroll"]*(kernel["MIWaveTile"][0] * numReadPerVectorA) // tensorParametersA["localReadInstruction"].numOffsets
+      numB = kernel["InnerUnroll"]*(kernel["MIWaveTile"][1] * numReadPerVectorB) // tensorParametersB["localReadInstruction"].numOffsets
+      readsPerIter = numA + numB if self.numReadsIterCoalesced == 1 else 0
+      readsLatency = roundUp(readsPerIter / numMfmaPerIter)*2
+      # number of local write instructions between 2 mfma
+      self.numLocalWriteModPerMfma = max((MatrixInstructionLatency - readsLatency)//(LocalWriteLatency+1),1)
+      ##################################
+      numGlobalReadInsPerIter = numMfmaPerIter * self.numGlobalReadInsPerMfma
+      numLocalWriteModPerIter = numMfmaPerIter * self.numLocalWriteModPerMfma
+      # if numGlobalReadInsPerMfma>1, we still want to schedule only 1 GlobalReadIncCode per mfma
+      # inserting empty CodeModule so that generator will schedule 1 GlobalReadIncCode 1 empty CodeModule if numGlobalReadInsPerMfma=2
+      numEmptyGlobalReadIncCode = self.numGlobalReadInsPerMfma - 1
```

Comment:
it would be good idea to mvoe inc() code into last iteration esp when PLR!=0. usually there nothing but xdlops  to schedule in last iteration math code

Reply:
if number of mfma per iteration is 1, it will cause bubbles if we move 2 inc() code to last iteration.
I'll try to break inc() code into small pieces and schedule it with localWrite.